core/src/main/java/org/apache/iceberg/rest/RESTUtil.java

```diff
@@ -215,8 +232,33 @@ public static String encodeNamespace(Namespace ns) {
    * @return a namespace
    */
   public static Namespace decodeNamespace(String encodedNs) {
-    Preconditions.checkArgument(encodedNs != null, "Invalid namespace: null");
-    String[] levels = Iterables.toArray(NAMESPACE_ESCAPED_SPLITTER.split(encodedNs), String.class);
+    return decodeNamespace(encodedNs, NAMESPACE_ESCAPED_SEPARATOR);
+  }
+
+  /**
+   * Takes in a string representation of a namespace as used for a URL parameter and returns the
+   * corresponding namespace.
+   *
+   * <p>See also {@link #encodeNamespace} for generating correctly formatted URLs.
+   *
+   * @param encodedNamespace a namespace to decode
+   * @param namespaceSeparator The namespace separator to use for decoding. This should be the same
+   *     separator that was used when calling {@link RESTUtil#encodeNamespace(Namespace, String)}
+   * @return a namespace
+   */
+  public static Namespace decodeNamespace(String encodedNamespace, String namespaceSeparator) {
+    Preconditions.checkArgument(encodedNamespace != null, "Invalid namespace: null");
+    Preconditions.checkArgument(
+        !Strings.isNullOrEmpty(namespaceSeparator), "Invalid namespace separator: null or empty");
+    String[] levels;
+
+    // for backwards compatibility
+    if (encodedNamespace.contains(NAMESPACE_ESCAPED_SEPARATOR)) {
+      levels = Iterables.toArray(NAMESPACE_SPLITTER.split(encodedNamespace), String.class);
+    } else {
+      levels =
+          Iterables.toArray(Splitter.on(namespaceSeparator).split(encodedNamespace), String.class);
```

Comment:
These branches are the same except for the splitter, right? It would be cleaner to create a splitter and then use the same line to split and convert to an array.

Reply:
yep good point, done